### ISSUE TITLE
Pretty print type values with -Z

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#2484]
Repro is with `zq` commit dd946d3.

Right now if `zq -Z` is used to pretty print ZSON output, each type value is still represented as one long string. For example, with the Zeek [zed-sample-data](https://github.com/brimdata/zed-sample-data):

```
$ zq -Z 'count() by typeof(.)' *
{
    typeof: ({_path:string,ts:time,peer:bstring,mem:uint64,pkts_proc:uint64,bytes_recv:uint64,pkts_dropped:uint64,pkts_link:uint64,pkt_lag:duration,events_proc:uint64,events_queued:uint64,active_tcp_conns:uint64,active_udp_conns:uint64,active_icmp_conns:uint64,tcp_conns:uint64,udp_conns:uint64,icmp_conns:uint64,timers:uint64,active_timers:uint64,files:uint64,active_files:uint64,dns_requests:uint64,active_dns_requests:uint64,reassem_tcp_size:uint64,reassem_file_size:uint64,reassem_frag_size:uint64,reassem_unknown_size:uint64}),
    count: 5 (uint64)
} (=0)
{
    typeof: ({_path:string,ts:time,uid:bstring,id:{orig_h:ip,orig_p:port=(uint16),resp_h:ip,resp_p:port=(uint16)},trans_depth:uint64,method:bstring,host:bstring,uri:bstring,referrer:bstring,version:bstring,user_agent:bstring,origin:bstring,request_body_len:uint64,response_body_len:uint64,status_code:uint64,status_msg:bstring,info_code:uint64,info_msg:bstring,tags:|[zenum=(string)]|,username:bstring,password:bstring,proxied:|[bstring]|,orig_fuids:[bstring],orig_filenames:[bstring],orig_mime_types:[bstring],resp_fuids:[bstring],resp_filenames:[bstring],resp_mime_types:[bstring]}),
    count: 144034
} (0)
...
```

To make it easier for users to see the shape of their data, we've envisaged being able to pretty print the type detail as well.

**Comment by philrz** [comment]
@mccanne has a branch where this was working fairly well, so he thinks he can revive that and get it merged.

**Comment by philrz** [comment]
I've verified that this was addressed as of Zed commit 268dfc1 which is associated with #3819.

```
$ zq -version
Version: v1.0.0-43-g268dfc13

$ zq -Z 'count() by typeof(this)' *
{
    typeof: <{
        _path: string,
        ts: time,
        uid: string,
        id: {
            orig_h: ip,
            orig_p: port=uint16,
            resp_h: ip,
            resp_p: port
        },
        duration: duration,
        version: string,
        community: string,
        get_requests: uint64,
        get_bulk_requests: uint64,
        get_responses: uint64,
        set_requests: uint64,
        display_string: string,
        up_since: time
    }>,
    count: 65 (uint64)
}
{
    typeof: <{
        _path: string,
        ts: time,
        ts_delta: duration,
        peer: string,
        gaps: uint64,
        acks: uint64,
        percent_lost: float64
    }>,
    count: 2 (uint64)
}
...
```

Thanks @mattnibs!
